### PR TITLE
Secure checkout complete action

### PR DESCRIPTION
**Pull request by lchrusciel**
This PR contains the improved implementation of #351. Current implementation will support following scenarios:

1. User is logged in - valid
2. User is not logged in and customer exists, but doesn't have an account - valid
3. User is logged in and customer exists and they are related - valid
4. User is not logged in and customer exists, but has an account - exception, he needs to log in
5. User is logged in but different mail is provided - exception

### CODE REVIEW

**Comment by lchrusciel** [comment]
/cc @mamazu 

Will it work for you?

**Comment by mamazu** [comment]
Thanks for picking up the work.

File: src/Exception/WrongUserException.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\Exception;
+
+use Exception;
+
+class WrongUserException extends Exception
```
Comment (Zales0123) [inline src/Exception/WrongUserException.php]: Missing `final` :(

File: src/Exception/WrongUserException.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\Exception;
+
+use Exception;
```
Comment (Zales0123) [inline src/Exception/WrongUserException.php]: Unneeded

File: src/Exception/WrongUserException.php (hunk 1 above)
Comment (Zales0123) [reply src/Exception/WrongUserException.php]: And `Exception` -> `\Exception`

File: src/Provider/ShopUserAwareCustomerProvider.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\ShopApiPlugin\Exception\WrongUserException;
+
+final class ShopUserAwareCustomerProvider implements CustomerProviderInterface
+{
+    /** @var CustomerRepositoryInterface */
+    private $customerRepository;
+    /** @var FactoryInterface */
+    private $customerFactory;
+    /** @var LoggedInShopUserProviderInterface */
```
Comment (Zales0123) [inline src/Provider/ShopUserAwareCustomerProvider.php]: Missing blank lines between properties

**Comment by Zales0123** [comment]
Thanks, Łukasz! :1st_place_medal:

**Comment by lchrusciel** [comment]
And Maximilian!

**Comment by Zales0123** [comment]
And Maximilian! 😄

**Pull request merged**